### PR TITLE
WT-11352 Update the architecture guide with background compaction

### PR DESCRIPTION
**Pull request by etienneptl**
- Tidied up the existing section about compaction
- Added a new section dedicated to background compaction

### CODE REVIEW

File: src/docs/arch-compact.dox [hunk 1]
```diff
@@ -18,15 +18,22 @@ the blocks at the end of the file to the reusable blocks at the start of the fil
...
 in the cache, and the intention here is to write them out and then discard previous checkpoints so
 that there are as many blocks as possible on the file's "available for reuse" list when the compaction
-starts.  The compaction logic then starts examining the files that need to be compacted. Compaction
-ignores files smaller than 1MB and computes the available bytes in the initial 80% and 90% of the file.
-Compaction doesn't proceed if at least 1MB worth of space can not be recovered. If at least 20% of
-the space is available in the first 80% of the file, the compaction process will try to rewrite
-the last 20% of the file. Else if at least 10% of the total file is available in the first 90% of
-the file, compaction is tried on the last 10% of the file. Else the file is skipped.
+starts. The compaction logic then starts examining the files that need to be compacted. Compaction
+does not proceed if one of the following conditions is met:
```
Comment (clarissecheah) [inline src/docs/arch-compact.dox:22]: Could we say: "...does not proceed if any of the following conditions are met:"

File: src/docs/arch-compact.dox [hunk 2]
```diff
@@ -18,15 +18,22 @@ the blocks at the end of the file to the reusable blocks at the start of the fil
...
+
+    - The file is smaller than 1MB.
+    - Less than the configured \c free_space_target (see \c WT_SESSION::compact) worth of space
+can be theoretically recovered.
+
+Once enough free space is available, one of the two following conditions has to be met for compaction to proceed:
+
+    - If at least 20% of the space is available in the first 80% of the file, the compaction process
```
Comment (clarissecheah) [inline src/docs/arch-compact.dox:30]: - At least 20% of the available space lies within the first 80% of the file (compaction will then try to rewrite the last 20% of the file).
- At least 10% of the total file is available in the first 90% of the file (compaction will then attempt to work on the last 10% of the file).

File: src/docs/arch-compact.dox [hunk 3]
```diff
@@ -62,4 +69,21 @@ track current progress. Some of the verbose logs for compaction can be activated
 verbose logs, WT also records statistics about compact progress that can be used to debug issues
 with the compaction process.
 
+# Background compaction #
+
+Background compaction is a WiredTiger service that runs compaction autonomously on all the eligible files
+present in the database while being the least disruptive as possible. Every file is eligible unless:
+
+    - it belongs to a tiered table or,
+    - is part of the exclude list that can be configured when enabling background compaction.
```
Comment (clarissecheah) [inline src/docs/arch-compact.dox:78]: np: it is *

File: src/docs/arch-compact.dox [hunk 4]
```diff
@@ -62,4 +69,21 @@ track current progress. Some of the verbose logs for compaction can be activated
...
+
+To limit disruption, background compaction reduces the number of checkpoints by skipping the first one
+described in the section above. Furthermore, background compaction gathers statistics over time to
+keep track of tables where compaction is successful. Using those statistics, the service focuses on
+tables where compaction is likely to reclaim space and avoids unnecessary attempts.
+
+Background compaction can be enabled, configured and disabled through the same API \c WT_SESSION::compact.
```
Comment (clarissecheah) [inline src/docs/arch-compact.dox:85]: np:
- comma after configured 
- ...as soon as possible and when re-enabled, it starts..."

File: src/docs/arch-compact.dox (hunk 1 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: Thank you, see 0b119b18000615d5915fd895b70ca723217c6b41.

File: src/docs/arch-compact.dox (hunk 2 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: Thank you, see 0b119b18000615d5915fd895b70ca723217c6b41.

File: src/docs/arch-compact.dox (hunk 3 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: Thank you, see 0b119b18000615d5915fd895b70ca723217c6b41.

File: src/docs/arch-compact.dox (hunk 4 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: Thank you, see 0b119b18000615d5915fd895b70ca723217c6b41.

File: src/docs/arch-compact.dox [hunk 5]
```diff
@@ -18,15 +18,22 @@ the blocks at the end of the file to the reusable blocks at the start of the fil
...
+can be theoretically recovered.
+
+Once enough free space is available, one of the two following conditions has to be met for compaction to proceed:
+
+    - At least 20% of the available space lies within the first 80% of the file (compaction will
+then try to rewrite the last 20% of the file).
+    - At least 10% of the total file is available in the first 90% of the file (compaction will then
+attempt to work on the last 10% of the file).
 
 Compacting the object is done 10% at a time, that is, compaction tries to move blocks from the last 10% of the
```
Comment (pmacko86) [inline src/docs/arch-compact.dox:35]: I think this should say "10% or 20% at a time"?

File: src/docs/arch-compact.dox [hunk 6]
```diff
@@ -62,4 +69,21 @@ track current progress. Some of the verbose logs for compaction can be activated
 verbose logs, WT also records statistics about compact progress that can be used to debug issues
 with the compaction process.
 
+# Background compaction #
+
+Background compaction is a WiredTiger service that runs compaction autonomously on all the eligible files
+present in the database while being the least disruptive as possible. Every file is eligible unless:
+
+    - it belongs to a tiered table or,
+    - it is part of the exclude list that can be configured when enabling background compaction.
```
Comment (pmacko86) [inline src/docs/arch-compact.dox:78]: "it is a part of" (you're missing an article)

**Review by pmacko86 — APPROVED** [review]
LGTM. Just two small comments.

File: src/docs/arch-compact.dox (hunk 5 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: You are right! Fixed in fe7a5bfc6aa1b1fbb3fd94285251c87a3b3361a5.

File: src/docs/arch-compact.dox (hunk 6 above)
Comment (etienneptl) [reply src/docs/arch-compact.dox]: Fixed in fe7a5bfc6aa1b1fbb3fd94285251c87a3b3361a5, thank you!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by jiechenbo — APPROVED** [review]
LGTM

**Pull request merged**